lib/src/contracts/contract_invoker.dart

```diff
@@ -0,0 +1,118 @@
+part of 'package:web3dart/contracts.dart';
+
+class ContractInvocation {
+  final Web3Client client;
+  final DeployedContract contract;
+  final ContractFunction function;
```

Comment:
I would make these parameters private:

```suggestion
  final Web3Client _client;
  final DeployedContract _contract;
  final ContractFunction _function;
```